packages/dev/core/src/Audio/v2/abstractAudioEngine.ts

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable babylonjs/available */
+/* eslint-disable jsdoc/require-jsdoc */
+
+import type { AbstractAudioDevice } from "./abstractAudioDevice";
+import type { AbstractAudioNode } from "./abstractAudioNode";
+import { AbstractAudioNodeParent } from "./abstractAudioNodeParent";
+import type { AbstractAudioPositioner } from "./abstractAudioPositioner";
+import type { AbstractAudioSender } from "./abstractAudioSender";
+import type { AbstractMainAudioBus } from "./abstractMainAudioBus";
+import type { AbstractSoundSource } from "./abstractSoundSource";
+import type { AbstractStaticSoundInstance } from "./abstractStaticSoundInstance";
+import type { AbstractStaticSoundSource } from "./abstractStaticSoundSource";
+import type { AbstractStreamingSoundInstance } from "./abstractStreamingSoundInstance";
+import type { AbstractStreamingSoundSource } from "./abstractStreamingSoundSource";
+import type { SpatialAudioListener } from "./spatialAudioListener";
+import type { Nullable } from "../../types";
+
+/**
+ * Owns top-level AbstractAudioNode objects.
+ * Owns all AbstractSoundSource objects.
+ */
+export abstract class AbstractAudioEngine extends AbstractAudioNodeParent {
+    // Owned
+    private _listeners: Nullable<Set<SpatialAudioListener>> = null;
+
+    // Not owned, but all items should in parent's `children` container, too, which is owned.
+    public readonly soundInstances = new Set<AbstractStaticSoundInstance>();
+
+    // Owned
+    public readonly soundSources = new Set<AbstractSoundSource>();
+
+    public override dispose(): void {
+        this.soundInstances.clear();
+
+        if (this._listeners) {
+            for (const listener of this._listeners) {
+                listener.dispose();
+            }
+            this._listeners.clear();
+        }
+
+        for (const source of this.soundSources) {
+            source.dispose();
+        }
+        this.soundSources.clear();
+
+        super.dispose();
+    }
+
+    public get listeners(): Set<SpatialAudioListener> {
+        if (!this._listeners) {
+            this._listeners = new Set();
+        }
+
+        return this._listeners;
+    }
```

Comment:
Are users of an audio engine expected to be able to modify the set? It seems creating a listener adds to this list, but it's also possible for someone to do this manually? Should the return type be `DeepImmutable`?

Same question for `soundInstances` and `soundSources`.